### PR TITLE
Tharper/kill instances via goal

### PR DESCRIPTION
**Pull request by timcharper**
<<Replace this line with your Revision Title>>

Summary:
For more details follow http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html.

JIRA issues:

### CODE REVIEW

File: src/main/scala/mesosphere/marathon/api/TaskKiller.scala [hunk 1]
```diff
@@ -69,7 +79,28 @@ class TaskKiller @Inject() (
     }
   }
 
-  private[this] def expunge(instances: Seq[Instance]): Future[Done] = {
+  private[this] def awaitConsideredTerminal(instanceIds: Set[Instance.Id]): Source[Set[Instance.Id], NotUsed] =
+    instanceTracker.instanceUpdates
+      .flatMapConcat {
+        case (snapshot, updates) =>
+          val pendingInstances: Set[Instance.Id] = snapshot.instances.collect {
```
Comment (jeschkies) [inline src/main/scala/mesosphere/marathon/api/TaskKiller.scala]: Hm, this is tricky. What about terminal instances with `Goal.Running`?

File: src/main/scala/mesosphere/marathon/api/TaskKiller.scala [hunk 2]
```diff
@@ -69,7 +79,28 @@ class TaskKiller @Inject() (
...
+            .filter { change => instanceIds(change.id) }
+            .scan(pendingInstances) {
+              case (remaining, deleted: InstanceDeleted) =>
+                remaining - deleted.id
+              case (remaining, InstanceUpdated(instance, _, _)) if considerTerminal(instance.state.condition) =>
+                remaining - instance.instanceId
+              case (remaining, _) =>
+                remaining
+            }
+            .takeWhile(_.nonEmpty)
```
Comment (jeschkies) [inline src/main/scala/mesosphere/marathon/api/TaskKiller.scala]: This is way easier to understand than the kill watcher 🙂.

File: src/main/scala/mesosphere/marathon/api/TaskKiller.scala (hunk 1 above)
Comment (timcharper) [reply src/main/scala/mesosphere/marathon/api/TaskKiller.scala]: That's a great catch; those instances will be restarted and we will need to wait for the goal and the status to be terminal

File: src/main/scala/mesosphere/marathon/api/TaskKiller.scala [hunk 3]
```diff
@@ -43,22 +47,28 @@ class TaskKiller @Inject() (
           val activeInstances = foundInstances.filter(_.isActive)
 
           if (wipe) {
-            val instancesAreTerminal: Future[Done] = killService.watchForKilledInstances(activeInstances)
+            val foundInstanceIds = Set(foundInstances.map(_.instanceId): _*)
```
Comment (meln1k) [inline src/main/scala/mesosphere/marathon/api/TaskKiller.scala]: Just wonder, why not 
```suggestion
            val foundInstanceIds = foundInstances.map(_.instanceId).toSet?
```

**Pull request closed without merging**